pkg/azuredisk/azuredisk.go

```diff
@@ -72,9 +72,9 @@ const (
 )
 
 var (
-	managedDiskPathRE       = regexp.MustCompile(`.*/subscriptions/(?:.*)/resourceGroups/(?:.*)/providers/Microsoft.Compute/disks/(.+)`)
+	managedDiskPathRE       = regexp.MustCompile(`(?i).*/subscriptions/(?:.*)/resourceGroups/(?:.*)/providers/Microsoft.Compute/disks/(.+)`)
```

Comment:
add `(?i)` means ignore case? and could you add some test cases into `azuredisk_test.go` in this PR?

Reply:
ut added